### PR TITLE
Fixes VPC selection to make sure it has internet gateway attached.

### PR DESCRIPTION
**Pull request by hsong-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1557377

### CODE REVIEW

**Comment by hsong-rh** [comment]
@roliveri @jerryk55 @bronaghs Please review.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 1]
```diff
@@ -362,6 +362,29 @@ def enabled_dns_support?(vpc)
...
 
+  def enabled_internet_gateways?(vpc)
+    igw_ids = vpc.internet_gateways.map(&:internet_gateway_id)
+
+    unless igw_ids.empty? # Gateway attached
+      _log.debug("Found a gateway [#{igw_ids.first}] on VPC [#{vpc.id}]")
+
+      # Make sure gateway has vaild route
+      route_tables = vpc.route_tables.select do |route_table|
+        route_table.routes.any? { |route| route.gateway_id == igw_ids.first }
```
Comment (jerryk55) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: Can't there be more than one gateway?  It looks like you're only checking if the route's gateway matches the first gateway id - what if the route's gateway matches another one of the gateways on the VPC?

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 2]
```diff
@@ -362,6 +362,29 @@ def enabled_dns_support?(vpc)
...
+      end
+
+      unless route_tables.empty?
+        _log.debug("Found route tables #{route_tables} on the gateway [#{igw_ids.first}]")
+        subnets = route_tables.map { |rt| rt.associations.first.subnet_id }
```
Comment (jerryk55) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: I guess here we only care if there is one subnet so this use of "first" is ok, right?

File: app/models/manageiq/providers/amazon/agent_coordinator.rb (hunk 1 above)
Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: It's a one to one relationship between VPC and gateway.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb (hunk 2 above)
Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: Yes, the API only has one entry per association returned.

**Comment by jerryk55** [comment]
👍 LGTM

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 431b61aa9a1df03234e45f924c5ba91e7497682b
Author: Richard Oliveri <oliveri.richard.github@gmail.com>
Date:   Wed Mar 21 14:33:00 2018 -0400

    Merge pull request #424 from hsong-rh/check_vpc_gateway
    
    Fixes VPC selection to make sure it has internet gateway attached.
    (cherry picked from commit 0f4f60b92327d9c5fa819518b5fd934035c29ac8)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1559609
```

**Pull request merged**